test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/EventTransformTest.java

```diff
@@ -56,33 +59,126 @@ public static void setup() {
     }
 
     @Test
-    public void testUnmarshalling() throws Exception {
+    public void testUnmarshalling_JsonBlobPayload() throws Exception {
         HttpResponseHandler<SdkPojo> responseHandler = protocolFactory
                 .createResponseHandler(JsonOperationMetadata.builder().build(), InputEvent::builder);
 
         InputEvent unmarshalled = (InputEvent) responseHandler.handle(SdkHttpFullResponse.builder()
-                        .content(AbortableInputStream.create(SdkBytes.fromUtf8String(EXPLICIT_PAYLOAD).asInputStream()))
+                        .content(AbortableInputStream.create(SdkBytes.fromUtf8String(EXPLICIT_PAYLOAD_JSON).asInputStream()))
                         .putHeader(HEADER_MEMBER_NAME, HEADER_MEMBER)
                         .build(),
                 new ExecutionAttributes());
 
         assertThat(unmarshalled.headerMember()).isEqualTo(HEADER_MEMBER);
-        assertThat(unmarshalled.explicitPayloadMember().asUtf8String()).isEqualTo(EXPLICIT_PAYLOAD);
+        assertThat(unmarshalled.explicitPayloadMember().asUtf8String()).isEqualTo(EXPLICIT_PAYLOAD_JSON);
     }
 
     @Test
-    public void testMarshalling() {
+    public void testUnmarshalling_NonJsonBlobPayload() throws Exception {
```

Comment:
Can we parameterize these tests since they're pretty similar? Perhaps one for Blob and one for String, or if there are enough similarities, one for all 4 unmarshalling tests. 
If you go with 2 tests, you may get a way with a `ValueSource` source for the args